domain/upgrade/state/state.go

```diff
@@ -200,72 +201,65 @@ func (st *State) SetControllerDone(ctx context.Context, upgradeUUID, controllerI
 	lookForDoneNodesQuery := `
 SELECT (controller_node_id, node_upgrade_completed_at) AS &infoControllerNode.*
 FROM   upgrade_info_controller_node
-WHERE  upgrade_info_uuid = $M.info_uuid`
-	lookForDoneNodesStatement, err := sqlair.Prepare(lookForDoneNodesQuery, infoControllerNode{}, sqlair.M{})
+WHERE  upgrade_info_uuid = $M.info_uuid
+  AND controller_node_id = $M.controller_id;`
+	lookForDoneNodesStmt, err := sqlair.Prepare(lookForDoneNodesQuery, infoControllerNode{}, sqlair.M{})
```

Comment:
There's still a number of vars suffixed `Statement` in this file. Would be nice to change over the rest whilst you're at it